libraries/Microsoft.Bot.Builder.Ai/QnAMakerMiddleware.cs

```diff
@@ -12,10 +13,23 @@ namespace Microsoft.Bot.Builder.Ai
     public class QnAMakerMiddleware : Middleware.IReceiveActivity
     {
         private readonly QnAMaker _qnaMaker;
+        private readonly QnAMakerMiddlewareOptions _options;
 
-        public QnAMakerMiddleware(QnAMakerOptions options, HttpClient httpClient)
+        public QnAMakerMiddleware(QnAMakerMiddlewareOptions options, HttpClient httpClient)
         {
-            _qnaMaker = new QnAMaker(options, httpClient);
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+
+            var qnaMakerOptions = new QnAMakerOptions()
+            {
+                KnowledgeBaseId = _options.KnowledgeBaseId,
+                SubscriptionKey = _options.SubscriptionKey,
+                ScoreThreshold = _options.ScoreThreshold,
+                Top = _options.Top,
+                MetadataBoost = _options.MetadataBoost,
+                StrictFilters = _options.StrictFilters
+            };
+
+            _qnaMaker = new QnAMaker(qnaMakerOptions, httpClient);
```

Comment:
Why gen-up the QnaMakerOptions, when you could just pass in the QnAMakerMiddlwareOptions? Because of the inheritence, that should work just fine.

Reply:
Good point.